### DISCUSSION TITLE
Pocketbase as a framework with Pocketbase Svelte UI

### DISCUSSION THREAD

**Discussion by ronny-rentner** [pocketbase/pocketbase#2059] [Q&A] [answered]
Hi,

I really like the concept of Pocketbase as a framework. I want to use it for a SaaS. I also want to use the built-in admin UI and extend it as a basis for my future users as it saves me from a lot of duplications. I think modern admin UIs should be built into the main product. If you log in as a user, you will see a restricted version if the UI limited to your scope.

Do you have any recommendations or thoughts regarding this approach? Any tips how to stay updatable or have the least complications?

Thanks in advance!

**Comment by ganigeorgiev** [comment, marked as answer]
The built-in Admin UI at the moment unfortunately is not customizable as it was intended only as a dev dashboard and for quick data exploration. It cannot be changed without forking the project but as you've already noted this may complicate future updates and you'll have to maintain it on your own.

Support for external addons and UI "hooks" eventually will be added as part of #898 sometime in the future.

Support for admin "roles" was also suggested previously and it is in my todo to consider it but for now is left out of the scope for v1 until the other more major tasks are completed.

For now, if you want to have an app level admin dashboard, my recommendation would be to create it as a separate frontend SPA customized and tailored for your specific use case. Similar to the built-in Admin UI, you can also embed it as part of the final executable using [`//go:embed`](https://pkg.go.dev/embed).

**Comment by ronny-rentner** [comment]
Thanks for your quick reply. I'll consider your recommendation. My strategy would have been to merge from upstream regularly and to try to avoid changing upstream files as far as possible. The changes would be limited to adding those hooks where needed. I expect my biggest challenge are actually those admin roles.

**Reply by nicosql** [reply]
> Thanks for your quick reply. I'll consider your recommendation. My strategy would have been to merge from upstream regularly and to try to avoid changing upstream files as far as possible. The changes would be limited to adding those hooks where needed. I expect my biggest challenge are actually those admin roles.

been doing this for 2 years it works fine, as long as you're disciplined and have your git routines down I see no issue - this is extremely common in my domain.  We have donated to Gani before but now donations are an aside which is fine, it's just paperwork on this side but I say that to say - go for it.
